### PR TITLE
feat: add auth

### PR DESCRIPTION
**Pull request by drewradcliff**

### CODE REVIEW

File: .github/workflows/fly-review.yml [hunk 1]
```diff
@@ -35,3 +35,5 @@ jobs:
           secrets: |
             UPSTASH_REDIS_REST_URL=${{ secrets.UPSTASH_REDIS_REST_URL }}
             UPSTASH_REDIS_REST_TOKEN=${{ secrets.UPSTASH_REDIS_REST_TOKEN }}
+            NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=${{ secrets.NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY }}
+            CLERK_SECRET_KEY=${{ secrets.CLERK_SECRET_KEY }}
```
Comment (lewxdev) [inline .github/workflows/fly-review.yml]: could alphabetize this just bc my ocd

File: app/layout.tsx [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
+    <ClerkProvider>
+      <html lang="en">
+        <body className={GeistSans.className}>
+          <div className="p-10">
+            <h1 className="text-4xl">mmmines</h1>
+            <SignedIn>
+              <SignOutButton />
+            </SignedIn>
+            <div className="pb-10" />
+            <div className="flex flex-col items-center justify-center">
```
Comment (lewxdev) [inline app/layout.tsx]: the centered div on mobile actually isn't the best experience, maybe we can update this

File: middleware.ts
Comment (lewxdev) [inline middleware.ts:1]: for the scope of this project, I don't think any routes should be protected, everything can be public, bit in order to interact with certain things the user would need to be authed
we can probably forego middleware?

File: .github/workflows/fly-review.yml (hunk 1 above)
Comment (drewradcliff) [reply .github/workflows/fly-review.yml]: 🫠

File: middleware.ts
Comment (drewradcliff) [reply middleware.ts:1]: clerk requires middleware but I will remove protected routes for now

File: app/layout.tsx (hunk 2 above)
Comment (lewxdev) [reply app/layout.tsx]: resolved in #15

**Review by lewxdev — APPROVED** [review]
ship it

**Pull request merged**